protocol/collectd/collectd.go

```diff
@@ -75,7 +75,13 @@ func NewDatapoint(point *JSONWriteFormat, index uint, defaultDimensions map[stri
 	addIfNotNullOrEmpty(dimensions, "dsname", !usedDsName, dsname)
 
 	timestamp := time.Unix(0, int64(float64(time.Second)**point.Time))
-	return datapoint.New(metricName, dimensions, datapoint.NewFloatValue(*val), metricType, timestamp)
+	var value datapoint.Value
+	if *val == float64(int64(*val)) {
```

Comment:
good eye. I tried that out in the benchmark test and had no effect on performance or allocations, so it gets optimized out and i felt the original was more readable.